### PR TITLE
SciPy global optimizers

### PR DESCRIPTION
**Pull request by onodip**
Implemented some of the global optimizers (basinhopping, differential evolution, simplicial homology global optimization, dual annealing) from SciPy in the `ScipyOptimizeDriver`. 
They have similar signature and result as `scipy.minimize`. The advantage would be, that OpenMDAO can be good in the local minimization for gradient calculation. 

For now, dual annealing and basinhopping have convergence problems with quite simple examples (so **some of the tests fail now**).

**TODO:**
- some more tests for bounds and constraints, for optimizers which support them
- tests for optimizer options (since the signatures of optimizers are not really consistent)
- fix convergence issues

### CODE REVIEW

**Comment by Kenneth-T-Moore** [comment]
Hey, Péter

Thank you for adding the other scipy optimizers. I think this looks good, except for the test failure. Is "Basin Hopping" the only failing method? Do you think this is an inherent problem (maybe it is not a good method for the type of problem), or is there something in the wrapper or on the openmdao side that can be changed to improve it and make the test pass? 

I'd recommend commenting out the test and the code for basinhopping for now; we wouldn't want someone to try to use the optimizer when it is not performing to expectations.

Also, we will probably need some updates to the scipy optimizer feature doc, but I think I might take care of that once it is ready to merge.

**Comment by onodip** [comment]
Most of the tests are based on the examples from the SciPy docs, so they should be appropriate problem for the optimizers. Now the tests do not fail. I reduced the bounds and increased the number of iterations for the problematic tests. So the issue is worse convergence than in the standalone SciPy, and can stuck more often in local minima. I could not isolate, what makes the difference in performance. 
The new basinhopping test (with the bounds) converged without problems.

**Comment by onodip** [comment]
Fails during package installations, seems unrelated to the PR.

File: openmdao/drivers/scipy_optimizer.py [hunk 1]
```diff
@@ -399,17 +409,86 @@ def run(self):
...
+
+                result = basinhopping(fun, x_init,
+                                      accept_test=accept_test,
+                                      **self.opt_settings)
+            elif opt == 'dual_annealing':
+                from scipy.optimize import dual_annealing
+                self.opt_settings.pop('disp')  # It does not have this argument
+                # There is no "options" param, so "opt_settings" can be used to set the (many)
+                # keyword arguments
+                # Note, that
```
Comment (swryan) [inline openmdao/drivers/scipy_optimizer.py]: leftover fragment of a comment?

Comment (onodip) [reply openmdao/drivers/scipy_optimizer.py]: Nice catch, deleted

File: openmdao/drivers/scipy_optimizer.py [hunk 2]
```diff
@@ -54,20 +54,11 @@
...
- pages = {37:1--37:39},
- articleno = {37},
- numpages = {39},
- doi = {10.1145/3182393},
- publisher = {ACM},
+@phdthesis{hwang_thesis_2015,
+  author       = {John T. Hwang},
+  title        = {A Modular Approach to Large-Scale Design Optimization of Aerospace Systems},
+  school       = {University of Michigan},
+  year         = 2015
```
Comment (swryan) [inline openmdao/drivers/scipy_optimizer.py]: It looks like you inadvertently reverted the citation change from the previous PR?

File: openmdao/drivers/tests/test_scipy_optimizer.py [hunk 3]
```diff
@@ -1497,6 +1497,301 @@ def test_multiple_objectives_error(self):
...
+        model.add_design_var('x', lower=-2.0 * np.ones(size), upper=-0.5 * np.ones(size))
+        model.add_objective('f')
+        prob.setup()
+        prob.run_driver()
+        assert_rel_error(self, prob['x'], -np.ones(size), 1e-2)
+        assert_rel_error(self, prob['f'], 3.0, 1e-2)
+
+    @unittest.skipUnless(LooseVersion(scipy_version) >= LooseVersion("1.2"),
+                         "scipy >= 1.2 is required.")
+    def test_shgo(self):
```
Comment (swryan) [inline openmdao/drivers/tests/test_scipy_optimizer.py]: Péter,

After adding a scipy 1.2 configuration to our Travis CI, we have found that this test does not pass (I was able to get it to pass locally by removing the bounds on the design variable).  We would appreciate it if you could take a look at your convenience.

Thanks!

Comment (swryan) [reply openmdao/drivers/tests/test_scipy_optimizer.py]: Also see Issue #909

Comment (onodip) [reply openmdao/drivers/tests/test_scipy_optimizer.py]: Hmm, for me the test passes with SciPy 1.2.1 on Windows, I will try to figure it out.

**Pull request merged**